packages/contracts-bedrock/scripts/libraries/DeployUtils.sol

```diff
@@ -18,6 +19,11 @@ library DeployUtils {
         require(_who.code.length > 0, string.concat("DeployUtils: no code at ", LibString.toHexStringChecksummed(_who)));
     }
 
+    function assertEIP1967Implementation(address _proxy) internal {
```

Comment:
nit, since we can use this for all proxy types
```suggestion
    function assertImplementationSet(address _proxy) internal {
```

Reply:
The reason I was explicit here was because we still have a `ResolvedDelegateProxy` and an `L1ChugSplashProxy`, both of which don't expose `implementation()`.

Reply:
Sorry, I was not clear. I was thinking (as needed) we can expand the function implementation to account for other proxy types. So first try `implementation()`, and if it reverts do `getImplementation`, etc.

Reply:
good call, makes sense!

Reply:
Updated in: https://github.com/ethereum-optimism/optimism/pull/11875/commits/42a049f9b90c6ae718c1651cf621af9bb7204e4b